### PR TITLE
Update to Rails 7.1

### LINKED ISSUE #1557: Update to Rails 7.1
Rails 6.x stops receiving security patches in October 2024.

According to https://docs.google.com/spreadsheets/d/1HoOh7Fjs_fz-5xSerTGfHIYsrYf-4QXhcv_ZvSVk9AM/edit?gid=515631444#gid=515631444 we should be running Rails 7.1 by the end of the year, anyways.

## Acceptance Criteria

- [ ] Application is running on Rails 7.1

## First Step

`bundle update`

### PR DESCRIPTION
**Pull request by tpendragon**
This is in draft, as I don't want us to have to debug anything that might go wrong with it right now.

Closes #1557

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -54,24 +53,26 @@ source "https://gems.contribsys.com/" do
...
 
+gem 'acts-as-taggable-on'
 gem 'bixby', '~> 5.0'
-gem 'blacklight', '~> 7.18', '< 7.25'
-gem 'blacklight-gallery'
+gem 'blacklight', '~> 7.18'
+gem 'blacklight-gallery', '~> 4.1.0'
 gem 'blacklight-oembed'
-gem 'blacklight-spotlight', '~> 3.0'
+gem 'blacklight-spotlight', '3.6.0.beta10'
```
Comment (cbeer) [inline Gemfile:61]: FWIW, Spotlight `3.6.0.beta` was rebranded as `4.0` because of the breaking changes introduced.

File: Gemfile [hunk 2]
```diff
@@ -56,7 +56,7 @@ end
 
 gem 'bixby', '~> 5.0'
 gem 'blacklight', '~> 7.18', '< 7.25'
-gem 'blacklight-gallery', '~> 4.4.0'
+gem 'blacklight-gallery', '~> 4.1.0'
```
Comment (hackartisan) [inline Gemfile:59]: Could we add a comment about what was bad when we tried to go higher than this?

File: config/environments/production.rb [hunk 3]
```diff
@@ -25,7 +25,11 @@
   config.public_file_server.enabled = true
 
   # Compress JavaScripts and CSS.
-  config.assets.js_compressor = Uglifier.new(harmony: true)
+  # config.assets.js_compressor = Uglifier.new(harmony: true)
```
Comment (hackartisan) [inline config/environments/production.rb:28]: delete commented out code?

File: Gemfile [hunk 4]
```diff
@@ -85,7 +86,7 @@ gem 'rsolr', '~> 2.0'
 # Required by blacklight-oembed
 gem 'ruby2_keywords'
 gem 'sidekiq', '~> 7.1.3'
-gem 'sir_trevor_rails'
+gem 'sir_trevor_rails', ">= 0.6.1", github: "madebymany/sir-trevor-rails"
```
[the remark this answers is not in the document]
Comment (tpendragon) [reply Gemfile:89]: :eyes: I missed that

File: app/controllers/catalog_controller.rb [hunk 5]
```diff
@@ -4,6 +4,7 @@
 # Simplified catalog controller
 class CatalogController < ApplicationController
   include Blacklight::Catalog
+  # include Spotlight::SearchHelper
```
[the remark this answers is not in the document]
Comment (tpendragon) [reply app/controllers/catalog_controller.rb]: Done.

File: Gemfile (hunk 2 above)
Comment (tpendragon) [reply Gemfile]: Turns out I can upgrade it it now, must have been me trying to get things to pass when there was that override that was necessary.

File: Gemfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (tpendragon) [reply Gemfile]: Done.

**Review by hackartisan — APPROVED** [review]
🎉

**Pull request merged**